src/test/ObjPropsDynamicSetter.Test.Unit/ObjectExtensionsTests.cs

```diff
@@ -0,0 +1,225 @@
+            yield return new object[] { new TestClass(), "InternalShortValue", -1654, true };
+            yield return new object[] { GetTestStruct(), "NestedTestStruct", new NestedTestStruct { NestedCharValue = 'c' }, false };
+        }
+
+        private static IEnumerable<object[]> GetValidTestClassDataForSetValue()
+        {
+            yield return new object[] { "ObjectValue", null, false };
+            yield return new object[] { "ObjectValue", 56, false };
+            yield return new object[] { "BoolValue", true, false };
+            yield return new object[] { "ByteValue", 24, false };
+            yield return new object[] { "SByteValue", 7, true };
+            yield return new object[] { "CharValue", 'U', false };
+            yield return new object[] { "StringValue", "test", false };
+            yield return new object[] { "UIntValue", 14, false };
+            yield return new object[] { "LongValue", 5786, false };
+            yield return new object[] { "ULongValue", 72424, false };
+            yield return new object[] { "UShortValue", 73, false };
+            yield return new object[] { "DecimalValue", 56.45632423m, false };
+            yield return new object[] { "DoubleValue", 82723.4563242d, false };
+            yield return new object[] { "FloatValue", 875.34f, false };
+            yield return new object[] { "DynamicValue", 56, false };
+            yield return new object[] { "NullableIntValue", null, false };
+            yield return new object[] { "NullableIntValue", 32, false };
+            yield return new object[] { "NullableDateTimeValue", null, true };
+            yield return new object[] { "NullableDateTimeValue", DateTime.Parse("1990-06-13", CultureInfo.InvariantCulture), false };
+            yield return new object[] { "EnumValue", TestEnumeration.Second, false };
+            yield return new object[] { "TestStruct", new TestStruct { IntValue = 12, ByteValue = 6, NestedTestClass = new NestedTestClass { NestedStringValue = "str" }, NestedTestStruct = new NestedTestStruct { NestedCharValue = 'l' } }, false };
+            yield return new object[] { "TestStruct.ByteValue", 5, false };
+            yield return new object[] { "TestStruct.NestedTestClass", new NestedTestClass { NestedStringValue = "internal string" }, false };
+            yield return new object[] { "TestStruct.NestedTestClass.NestedStringValue", "some string", false };
+            yield return new object[] { "TestStruct.NestedTestClass.PrivateIntValue", -78954, true };
+            yield return new object[] { "TestStruct.NestedTestStruct", new NestedTestStruct { NestedCharValue = '!' }, true };
+            yield return new object[] { "TestStruct.NestedTestStruct.NestedCharValue", ' ', false };
+            yield return new object[] { "ProtectedNestedTestClass.PrivateIntValue", 8347, true };
+            yield return new object[] { "IntCollectionValue", new int[] { 9, 3 }, false };
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Molnix888/obj-props-dynamic-setter/pullRequest?prid=6241271)